metrics-graphite/src/main/java/com/codahale/metrics/graphite/Graphite.java

```diff
@@ -181,6 +181,11 @@ public void close() throws IOException {
         } finally {
             this.socket = null;
         }
+        
+        // if hostname is used, re-resolve DNS in connect() by clearing the address
+        if (this.hostname != null) {
```

Comment:
I think if you specify a hostname and a port in the constructor, `Graphite` doesn't cache `InetSocketAddress` and it will make a DNS lookup everytime the the `connect` method is invoked.